pyvrp/cpp/Solution.cpp

```diff
@@ -139,16 +235,31 @@ Solution::Solution(ProblemData const &data,
 
     // Only store non-empty routes
     routes_.reserve(routes.size());
-    for (size_t idx = 0; idx != routes.size(); ++idx)
-        if (!routes[idx].empty())
-            routes_.emplace_back(data, routes[idx]);
+    for (auto const &route : routes)
+        if (!route.empty())
+            routes_.push_back(route);
+
+    if (data.isHeterogeneous())
+    {
+        // We sort routes by vehicle types. Combined with a stable sort, this
+        // ensures we keep the original sorting as much as possible.
+        auto comp = [](auto &a, auto &b) {
+            // If same type, empty vehicles first
```

Comment:
I forgot to remove this comment. I don't check for emptyness here since I would not like to support empty routes. But anyway I removed this sorting altogether.